data/json/mapgen/homeimprovement_superstore.json

```diff
@@ -0,0 +1,291 @@
+[
+    {
+        "type"  : "item_group",
+        "id"    : "pwr_clothing",
+        "items" : [
+            ["hat_hard",         100],
+            ["boots",            35],
+            ["glasses_safety",   50]
+        ]
+    },
+        {
+        "type"  : "item_group",
+        "id"    : "pwr_tools",
+        "items" : [
+            ["flashlight",       35],
+            ["wrench",           50],
+            ["screwdriver",      35]
+        ]
+    },
+        {
+        "type"  : "item_group",
+        "id"    : "pwr_parts",
+        "items" : [
+            ["cable",            65],
+            ["power_supply",     50],
+            ["amplifier",        50]
+        ]
+    },
+	    {
+        "type"  : "item_group",
+        "id"    : "wood",
+        "items" : [
+            ["2x4",            80],
+            ["power_supply",     50],
+            ["amplifier",        50]
+        ]
+    },
+    {
+        "type" : "mapgen",
+        "om_terrain" : ["hdwr_large_entrance"],
+        "method": "json",
+        "weight": 100,
+        "object": {
+            "rows":[
+            ".|s....................|",
+            ".|t.............rrrrrrr|",
+            ".......................|",
+            ".................ccc...|",
+            ".---.....r|r...........|",
+            ".|CT.....r|r.....ccc...|", 
```

Comment:
Nope, there is *no* terrain mapping for the letter `C`, what mapping you defined for any other letter does not matter at all. Look at the `terrain` object below, it contains mappings for `_ . , r - | + d W` but not for `C`, and also not for `^`, how is the game supposed to known what terrain those places should have? As I said before, the terrain *must* be specified for every square, furniture and the other stuff is optional, but you need a terrain everywhere.

@sparr The mapgen objects are loaded later (they need to know the other stuff for valid terrain etc. ids) and may be loaded in a different order. An error with that message is thrown when the terrain mapping contains a letter that is not mapped. But there is bug: the error is throw from the wrong json object and that makes the line wrong, but this only affects the display of the error.